### PR TITLE
Add Taylor–Green NS benchmark studying order reduction induced by time-dependent bcs

### PR DESCRIPTION
**Pull request by Ouardghi**
This PR adds a 2D Taylor–Green incompressible Navier–Stokes example implemented
with FEniCSx, together with a small runnable example and tests.

The example uses a monolithic velocity–pressure formulation and includes:
- time-dependent and time-independent Dirichlet boundary conditions,
- periodic boundary condition,
- a manufactured analytical solution for verification.

The main purpose of this benchmark is to study the effect of time-dependent
boundary conditions on numerical accuracy and order reduction in time
integration methods.

## Added

- Monolithic NSE formulation and Taylor–Green benchmark setup with  periodic and non-periodic bcs variants
- Small run example
- Convergence and correctness tests

### CODE REVIEW

**Comment by pancetta** [comment]
Does this setup give you the correct orders of accuracy as shown in the figure (the one you provided for a talk)?

File: pySDC/projects/StroemungsRaum_FEniCSx/problem_classes/NavierStokes_2D_TaylGreen_monolithic_FEniCSx.py [hunk 1]
```diff
@@ -0,0 +1,701 @@
...
+        Pe = basix.ufl.element(family, domain.topology.cell_name(), order - 1, shape=())
+        We = basix.ufl.mixed_element([Ve, Pe])
+        self.W = dfx.fem.functionspace(domain, We)
+        self.V, WV_map = self.W.sub(0).collapse()
+        self.Q, WQ_map = self.W.sub(1).collapse()
+
+        # get number of dofs and set up the problem
+        self.tmp_u = dfx.fem.Function(self.W)
+        nx = len(self.tmp_u.x.array)
+        print('DoFs on this level:', nx)
```
Comment (brownbaerchen) [inline pySDC/projects/StroemungsRaum_FEniCSx/problem_classes/NavierStokes_2D_TaylGreen_monolithic_FEniCSx.py:123]: Please change to debug level output with the logger

File: pySDC/projects/StroemungsRaum_FEniCSx/problem_classes/NavierStokes_2D_TaylGreen_monolithic_FEniCSx.py [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+
+        # invoke super init, passing number of dofs, dtype_u and dtype_f
```
Comment (brownbaerchen) [inline pySDC/projects/StroemungsRaum_FEniCSx/problem_classes/NavierStokes_2D_TaylGreen_monolithic_FEniCSx.py:125]:
```suggestion
```
this comment is lying

File: pySDC/projects/StroemungsRaum_FEniCSx/tests/test_Navier_Stokes_TaylGreen_monolithic_FEniCSx.py
Comment (brownbaerchen) [inline pySDC/projects/StroemungsRaum_FEniCSx/tests/test_Navier_Stokes_TaylGreen_monolithic_FEniCSx.py:1]: Would it be reasonable to add a check for the order here or would that take too long to include in the CI?

File: pySDC/projects/StroemungsRaum_FEniCSx/tests/test_Navier_Stokes_TaylGreen_monolithic_FEniCSx.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    w = prob.solve_system(rhs=b, factor=factor, u0=prob.dtype_u(prob.init), t=t)
+    w_periodic = prob_periodic.solve_system(rhs=b, factor=factor, u0=prob.dtype_u(prob.init), t=t)
+
+    # get the exact solution at time t for error computation
+    wx = prob.dtype_u(prob.init)
+    wx[:] = wex.x.array[:]
+
+    # Compute the relative monolithic velocity–pressure error for both non-periodic and periodic cases
+    rel_err = abs(wx - w) / abs(wx)
+    rel_err_periodic = abs(wx - w_periodic) / abs(wx)
```
Comment (brownbaerchen) [inline pySDC/projects/StroemungsRaum_FEniCSx/tests/test_Navier_Stokes_TaylGreen_monolithic_FEniCSx.py:57]: Shouldn't the exact solution be different for the two different BC implementations? If not then is the test meaningful?

File: pySDC/projects/StroemungsRaum_FEniCSx/environment.yml
Comment (brownbaerchen) [inline pySDC/projects/StroemungsRaum_FEniCSx/environment.yml:1]: Please add a new line for this new project in the CI file, i.e. just below [here](https://github.com/Parallel-in-Time/pySDC/blob/c5f2ed1419f1ba816be19778532c04923761d0f8/.github/workflows/ci_pipeline.yml#L102).
This is needed for the tests to be run at all.

File: pySDC/projects/StroemungsRaum_FEniCSx/run_Navier_Stokes_equations_TaylGreen_monolithic_FEniCSx.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    problem_params['t0'] = t0
+    problem_params['nelems'] = [64]
+    problem_params['family'] = 'CG'
+    problem_params['order'] = [2]
+
+    # initialize controller parameters
+    controller_params = dict()
+    controller_params['logger_level'] = 20
+
+    # Fill description dictionary for easy hierarchy creation
```
Comment (brownbaerchen) [inline pySDC/projects/StroemungsRaum_FEniCSx/run_Navier_Stokes_equations_TaylGreen_monolithic_FEniCSx.py:58]:
```suggestion
```

**Review by brownbaerchen — CHANGES_REQUESTED** [review]
Thanks a lot! This is a lot of stuff!

I didn't check the math because it's quite a lot :D 
I do agree that a script that shows the order reduction with Dirichlet BCs and not with periodic BCs would be good. If this can be run fast, ideally this would be a test as well.
Otherwise, we have to add the new project to the CI and I marked a few aesthetic things.

**Comment by pancetta** [comment]
Any update here?

**Comment by pancetta** [comment]
Superseded by #684
